.github/workflows/ci.yml

```diff
@@ -13,37 +13,40 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        cabal: ["3.2"]
+        cabal: ["3.4"]
         ghc:
-          - "8.10.1"
-          - "8.10.2"
+          - "9.0.0.20201227"
 
     steps:
     - uses: actions/checkout@v2
-      if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/ghc-8.10'
+      if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/ghc-9.0'
 
-    - uses: actions/setup-haskell@v1.1.4
+    - uses: haskell/actions/setup@v1
       id: setup-haskell-cabal
       name: Setup Haskell
       with:
         ghc-version: ${{ matrix.ghc }}
         cabal-version: ${{ matrix.cabal }}
 
-    - name: Freeze
+    - name: Prepare environment
       run: |
-        cabal freeze
+        echo "$HOME/.ghcup/bin" >> $GITHUB_PATH
+        sudo apt-get install libtinfo-dev
 
-    - uses: actions/cache@v1
+    - name: Freeze
+      run: |
+        cabal freeze --allow-newer
+ 
+    - uses: actions/cache@v2
       name: Cache ~/.cabal/store
       with:
         path: ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
-        key: ${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
+        key: ${{ runner.os }}
 
     - name: Build
       run: |
-        cabal configure --enable-tests --enable-benchmarks --test-show-details=direct
-        cabal build all
+        cabal configure --enable-tests --enable-benchmarks --test-show-details=direct --allow-newer -w ghc-9.0
```

Comment:
Likewise, instead of `--allow-newer` and `-w ghc-9.0` here, it’d be more convenient for building locally if this was specified in `cabal.project` (#1305)

Reply:
Okay great :)